<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>MBMethPred introduction</title>
<style type="text/css">
/**
 * Prism.s theme ported from highlight.js's xcode style
 */
pre code {
  padding: 1em;
}
.token.comment {
  color: #007400;
}
.token.punctuation {
  color: #999;
}
.token.tag,
.token.selector {
  color: #aa0d91;
}
.token.boolean,
.token.number,
.token.constant,
.token.symbol {
  color: #1c00cf;
}
.token.property,
.token.attr-name,
.token.string,
.token.char,
.token.builtin {
  color: #c41a16;
}
.token.inserted {
  background-color: #ccffd8;
}
.token.deleted {
  background-color: #ffebe9;
}
.token.operator,
.token.entity,
.token.url,
.language-css .token.string,
.style .token.string {
  color: #9a6e3a;
}
.token.atrule,
.token.attr-value,
.token.keyword {
  color: #836c28;
}
.token.function,
.token.class-name {
  color: #DD4A68;
}
.token.regex,
.token.important,
.token.variable {
  color: #5c2699;
}
.token.important,
.token.bold {
  font-weight: bold;
}
.token.italic {
  font-style: italic;
}
</style>
<style type="text/css">
body {
  font-family: sans-serif;
  max-width: 800px;
  margin: auto;
  padding: 1em;
  line-height: 1.5;
  box-sizing: border-box;
}
body, .footnotes, code { font-size: .9em; }
li li { font-size: .95em; }
*, *:before, *:after {
  box-sizing: inherit;
}
pre, img { max-width: 100%; }
pre, pre:hover {
  white-space: pre-wrap;
  word-break: break-all;
}
pre code {
  display: block;
  overflow-x: auto;
}
code { font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace; }
:not(pre) > code, code[class] { background-color: #F8F8F8; }
code.language-undefined, pre > code:not([class]) {
  background-color: inherit;
  border: 1px solid #eee;
}
table {
  margin: auto;
  border-top: 1px solid #666;
}
table thead th { border-bottom: 1px solid #ddd; }
th, td { padding: 5px; }
thead, tfoot, tr:nth-child(even) { background: #eee; }
blockquote {
  color: #666;
  margin: 0;
  padding-left: 1em;
  border-left: 0.5em solid #eee;
}
hr, .footnotes::before { border: 1px dashed #ddd; }
.frontmatter { text-align: center; }
#TOC .numbered li { list-style: none; }
#TOC .numbered { padding-left: 0; }
#TOC .numbered ul { padding-left: 1em; }
table, .body h2 { border-bottom: 1px solid #666; }
.body .appendix, .appendix ~ h2 { border-bottom-style: dashed; }
.footnote-ref a::before { content: "["; }
.footnote-ref a::after { content: "]"; }
.footnotes::before {
  content: "";
  display: block;
  max-width: 20em;
}

@media print {
  body {
    font-size: 12pt;
    max-width: 100%;
  }
  tr, img { page-break-inside: avoid; }
}
@media only screen and (min-width: 992px) {
  pre { white-space: pre; }
}
</style>
</head>
<body>
<div class="include-before">
</div>
<div class="frontmatter">
<div class="title"><h1>MBMethPred introduction</h1></div>
<div class="author"><h2></h2></div>
<div class="date"><h3></h3></div>
</div>
<div class="body">
<h2 id="input-file-for-prediction">Input file for prediction</h2>
<p>The <code>ReadMethylFile</code> is a function for reading DNA methylation beta values files and use them as new data for prediction by every model. The input for this function should be either CSV or TSV file format. Please uncomment the following lines and run the function.</p>
<h3 id="usage_1">Usage</h3>
<pre><code class="language-r"># set.seed(1234)
# fac &lt;- ncol(Data1)
# NewData &lt;- sample(data.frame(t(Data1[,-fac])),10)
# NewData &lt;- cbind(rownames(NewData), NewData)
# colnames(NewData)[1] &lt;- &quot;ID&quot;
# write.csv(NewData, &quot;NewData.csv&quot;, quote = FALSE, row.names = FALSE)
# methyl &lt;- ReadMethylFile(File = &quot;NewData.csv&quot;)
</code></pre>
<p>This function has only one argument, the File. The first column of the File is the CpG methylation probe that starts with cg characters and is followed by a number (e.g., cg100091). Other columns are samples with methylation beta values. All columns in the data frame should have a name.</p>
<h2 id="box-plot">Box plot</h2>
<p>The <code>BoxPlot</code> function draws a box plot out of DNA methylation beta values or other data frames.</p>
<h3 id="usage_2">Usage</h3>
<pre><code class="language-r">
data &lt;- Data2[1:20,]
data &lt;- cbind(rownames(data), data)
colnames(data)[1] &lt;- &quot;ID&quot;
BoxPlot(File = data, Projname = NULL)
</code></pre>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAkAAAAFoCAMAAACIU7TCAAAADFBMVEUAAAApgLkzMzP///97IkrPAAAACXBIWXMAAAsSAAALEgHS3X78AAAQTklEQVR4nO2diZLrKAxF6fD//zw1L7HNIkAs3uDcqjfGCIEgpzHGzsRYhDpk7g4AvVsAhLoEQKhLAIS6BECezK7a8uZ3fmJwj9RyHc7LhERUlpe85oZq6s7Va/+w1QD5KQBaXM6Hrfrcw/IAtLgAqFZTd65eAkDxClmmzAPIddIvyd+oibvWopgNb4XsHsTyohMAraNwUfyjxbvHcnGQF9GeE5ewlRTdxfsQ/H/weJBv42OneTV15+qV3NdJXI/EjcTI6ZLQb9LUnatX8q7KWdwkyiedAGghheubEKCQBgCaunP1KgHEDBRq6s7VC4BqNXXn6lUASLgLE/wBaF15i2hvZ3DPMiFksT+38cvK3RiM9wRzG4nuORuJyyp+H0j9KMM9j2o4O+77NHHXWhTis2X9UtY72iRA1qsCgBBKCYBQlwAIdQmAUJcACHUJgFCXAAh1CYBQlwAIdQmAUJeaAII6tAmAUJcACHUJgFCXAAh1CYBQlwAIdSnDgvx6cMEJLaYkCz4uZj9kndByUsxAAITSAiDUJQBCXaoH6PP5PBOgv7sDWFIvnYEkWADoDl0L0LDPeC6Aror8jHZKAHnfDO/eBwIgUX9i8tR2hunanWgAEgVAWrX0QPR5MEANYegB6uvjwwDqG6k+HwBq0Z+Q6tUQgNThAFDJpQ8gtRmAsqnOFgsaP0m0ANTSWwASMsUozhgpbZUApHVSh9MX9zMAOnWZAkD54gB0QuUnAKReWLqZAFQdxfh1LgDlizcU1GdOC5C63y1tA1C+8lEAFSov+Ih5AGQBSOkj5gGQBSBlaGIeANnpAeqrfAaAKpoBoNGVq73FPADKtvcQgBrifRtAo8zLAPT3k6pFAGoH6M8f6ScA1PJpxwBJh6oq8wEBUJBxO0BN82XaB4Cq2m4wB0MLQGkB0IMA8i+ZAARAh5MOIO8IQAB0OAFQn1nMA6BAIwHKeQPQyNAASFulqHcCpNgJA6B0JgApRuD3IUj3NWX8AKhQpTq0pwGkHwH/Q/gTygNQe5Xq0ADIPQKQzgxAZwEUPlwL9BqATiAEgFLyAIrjdAVASbOYB0CBAChpFvPaKwegawBKPZz7U5kVkQNQWSqAMtEWYpTqKXnrAUoEFI1P62NgACprDYAaI7/LPAdAcbQDX+dYB6CWUQOgNwB00RKpBd5ZAcqbUymbL3gfQHnvVOY7AZIfmgXVApBYDwD5BQ+U1gVIhwAAiQX/wtTjAao1A9BVAI3+XtjrACpsJO6nLQCNWmOfAFDqIgRAmYNY8FSABHMw4rcBtB86AQraXw+gREDnASQXeSxAB/AAlCkIQEmAdjMAZQoC0L0AifOcHTQULWanh9pFTsEMQKcCJLViBw1Fi7kwPlFBAAIgL1W4S7XhSN0CkPouH4CijDaA9I9i3wFQwbvYBwAKzalUtXdhfKKCjwSoWBCAQnMq9SiA/rqmt2UB2n9j15h/yd/h6yS3coyz9iYtlXoWQE4X6ysXvcVr81wAGc9ovJIpgPJBWO1QxOXnA0iq53KA4qn1LIB8mgBoFoCi0E4FyP3d+BMAkp45ApDfzFsB8kD6fD7nACQcAMjNzz/Vfw1A9qwZSDgsAFBhHSJ2Mdf2cwFyMmoAym8FisPnmK10lvd+G0CRuRUg7QR1PUAmzmgZAC+VHN0qgBraBqCLAPrtA7nYtCyiCz0EoHEAHd7aTaYSQOUJfuxOdMMAzABQYaP6BoDy3hMAJG5UvxegBvM7ACqHdhdAee/IE4CSBykfgEJPAEoepHwACj0BKHmQ8gEo9ASg5EHKB6DQE4CSBykfgEJPAEoepHwACj0BKHmQ8p3Ku174ASDPZ0mA8pUDkLrtKAoASprzu7wAlHg9AIC2FADJPpG3mAKgqtAAKEwBEAAlfGLvzFsPAARAZYAyQT4boGEv/HSZvxr5v3cBoKsA0poBSKpHND8NoNwAARCXsDJAeW8AqmllJoCE1QcAVQuAAAiA2gASfEqESO/cA1BNKwAEQL7uBajlDtBLXQxQrRmAzgaooW0vBUBDzF8BEAABEAAlzACUqQeAAAiAAAiAzgVI8T84AKDQO/K8B6D841CxnhMAUncRgMJSNl8PAPUD1PWtn68mBCj/1nxD5QAEQF7qWQB1f4jVAGl38h8DkH9yOkCSjx8LAAGQtu3nA5TvwxkADWwbgDSVi8/TxXqWBuj3izxWIQACoAigLz9WRdBsAFW/sgNAAkDOv5KmAyiqR4FAfptoSYBME0DlvfbtJPtn/jqATjMrPkRpp+JugKzRL4IAaD6AtB9OGiDpNCEuYTcDJPXjfoD+n30uAii/lXUcFgOo7UMcA1C9+SvvNl55EwZAABQDxF3Y/QBlCwIQAI0HqO9J2liAzHErX9JAgKQBOA43ACRcXZ8M0I3mr7zbeHP9owwAyvehyZxfYp4FUIW4hAEQAE0M0DXmr5w1kP4aBkAAFAG0nbtE7YnvU7LWH931TwBoaoCscY7GzWn/2W//pB8g9SVeemQkVQ5A4wAyMwEkeUuVA9AAgMIlkANQ/Avy5wIkEuJVUD0AfiwANML8VXLN7O9KmyPj8/mcDZBk9iro8I4zAOhagOz5M5Bk9iro8I4zAKgfIOPIMRg3BUAAlARIEADdZX4MIW0ABftAzgbQZftAeoDUz6L9WABohPmr6GHqA3aiAegV5q/C2/YnfLFQD1CDt5OR2U8CIFv8+/zqiS+UAdAzzMd/AciPxc8AoCEAfb/X3Pe9MOGlcEcANDdA0mlCJtEKAI0wtyIgjP71AOmUAijZStIMQMMA0nXsJIC0L0PvTgAEQF/te4M1DL0doHI9KwMkvUWVXp240DR9KyMKf26ALvjSxs0AuTVUAvTvvPYuLG72iQB5f1VBcDWfcRTMKgAJ3l95vMw7A0UDU6gHgKSKswDVLIIAaGGAgsl8kbuwqwAasUR6NkC+OXhvXisAAqBNr9lIHLXRGsUzFKC8NwDtTjcANMgcxQNAAARAAPQ+gE57m+gvXzkAARAAFQHK7PzOAlCDtx6gBrMuNAB6C0DtExQA2bBfabPbheEvRAHQXzSokwB0jK7TwwaAGsfHP5n5EjYpQIUY1cPX4h3Fcw9ADavkFoAaKndrAKDYO4rn4QCpp2hdaAA0CUDaLgLQIQBqAUht1oUGQA8EKD81AFCs6wDK30WUvG2YCUAA9ECA1N4AtDtdBZANMwEIgIoxAhAA+c3ngwCgTBcB6BAAAdDuBEAA9BUAARAA1XhH8QQFhYdUAARAaoCiigEoefgnAAIgAKrxjuIBIACq8Y7iOR2g+IUMADoFoPpXWgttPwMgoYsAdAZAsWk+gARzy2vLACTFGJuWAMg118/BAJSJdjmAdJUDUCLG2ARAAOR6XQ+QtLyI4gGg+QDSfuel0DYAAdBhaAFI8o7iAaD5ANpPxwG0z0VRPA8HSH2XrwvtQoCcn0g1v19O3XIA6DKA9F3UhXYdQOKvNu+21wGUjmcVgLSbTH5AIwD6nQEQADUDtF3C9lMAeh1Aee9zAfJAsgBUCvKNAHmLxMEAeVcy+/l81gBI2kyI2p0FID+MsQCZMOOeGUi4hhfa1gOU//sTzQD0FQCVAQo8AUgF0Lb7c2wALbaITqUASAlQRgA0CUDqmReA0vEsDJDYtpgCoHQ86wHkzTsAlG77eQAldgaqAWrfYch0EYBC8+MAkrz1ZukgA5R/HabQMTEFQOl45gNINANQonIAykQOQIcBgAAoEUQ1QNJdRNXbCplmCqEdSce70HbeW29uAUh6n7HQMTE1FUD93pmChcqPZG4EHgNQwVtsu8r7WoAqtz0BCIC8D6w22laAyjexmWYKlR9JAPpqPoCisAAoeZDarvIGoLCZlsoLkau9dZUDkDZaAAIgAAIgSQD0GIC63+gBoHTlyW20TDP6yqUzANLpJQCJocWFXg2QOvKXAeT/YbRFC0DrAhS2EgcGQNVmAAKgIyP/WqlcubDcASAptQBALZVLPgAkpQAIgIJW4sAACICOqrZ6E3sSUogABEBHVYVwxLyLARK2FZxm1NtxR3IxgJJzwyIASfEU7oQKla8FUKpgO0CJwX8TQK3eOoDG0yn5vBagylay5ikBOqFyyQeApNSLAaqZmAFI30rWPBFA/aEBkNhK1pwDqOZ7TQAEQH7Gn5D5HIC6VsmlysUzwed0gCQrAOULDkGgzwxANUHEfZX+uAFI8qkAqLhXOg9AYiYAST56gBrNkhWA8gX7xhmAFFoQoJKPuu0GgDKVJwGSXAAoOqi9E/GUzKKuACjxTvn8AMkN5oN4G0CSd6HKUR8iAEnJaQES8wpmdeb8AEkCoNEAjXqJL5mZL/gIgArmPoDyeU3YNFR5AkAtlb8foIKyANW/NJisvJCpFgC9CKCyWa6ovqBeLVW2zNb5zBaACmYAUuYBEAABUB9AYsm3AuRoPEAn6BUAjWr7ZQAdehtAJ/jkvfU1PgygiwRAw2ocBZCrNwEkZgLQRQAlVPrRXSfl/ujuhXowQC26E6AzqlT+7LcJfvb7bj2YkMt171i8FCB0CIDQiwVAqEv1AH0+HwBCm5iBUJcACHWptA9k9tRd+0Do0Xr+TjR6tAAIdQmAUJcACHUJgFCXAAh1qQ2gf/qYTVIK82TmqGAHQF99sinMk5nFggCEWWseDxBCAIQ61QWQiRLF+kR77G5EsxHyKtqJ3eVmhnfHyOa3jVq+mWr9vw7/Pa431oSp763aXjB0cXKN3evZU6KPU7mXjJpx2xGacVuUQhO749x2yD7H6MbdEUN72aiJzdhOgJwgTJDnFZRGyu1L7G2sE23UjNiOPFJCM26VQjNt3RE/uriZ946a2IztAcgOq2ZY2+IfxvPNVzXcW0GhYK08OIXZzZ9Px5rF64R4jXqF+ZpBEz+x4LNLm8WCtg+g7FTuJqVUl7lrKKrMJkyJmZ3m42M4klKq19wTuVjQ9gDkObfMqJhfZi4UrNXjLgSYzzSLBW0XQJddRjA/wDx+DYSQ7QLIX1WFKcyTmcWCzECoU10AmWwK82TmQsFaPfeGAfMJZrGg7QJovxJKKcyTmcWCtgcghGznDJRJYZ7MLBa0fQDZfXaLU5gnM4sFLZcw1CkuYZhVZi5hmLvMYkHbOQPtdVffFGJ+m1ksaHsAsi6GQgrzZGaxYCdAJpPCPJlZLMglDLPOLBa0AIRZaRYL2i6AHvvcD/MJZrGg7QHIqUVKYZ7MLBbsAgihvkuY3S6PQgrzZGaxoO0EyJpkCvNkZrGgm6jXsRSXUpgnM4sFuwBCCIBQpwAIdQmAMnJ3XOUCXsqYosN8Wq2/NfofhjwQAUCb00parLtVKgMhALTaiC7W3So5GyBmu0T9iDLbYbcd5dca0rV6W6djC+T7z5h/tGwUmS3zlwIgFGhbE28ABUnjp9yN2nW0Vm+r5bwEEwFkjZ8CICTJ2O0NKhmgr5FLGIq0L6LlS5gzA7EGQoI2gIwMUGINtNiILtbdKv3u3d2bLeeGLHEXttgEBEA5bY8mfjfvHkDSPhCPMpBSDNsmRqJJDNsmRqJJDNsmRgJ1CYBQlwAIdQmAUJcACHUJgFCX/gM13qpC2TcOUwAAAABJRU5ErkJggg==" alt="plot of chunk unnamed-chunk-3" /></p>
<p>This function has two arguments as follow:</p>
<ul>
<li><code>File</code>    A data frame with the first column as ID.</li>
<li><code>Projname</code> A string to name the plot.</li>
</ul>
<h2 id="t-sne-3d-plot">t-SNE 3D plot</h2>
<p>The <code>TSNEPlot</code> function draws a 3D t-SNE plot for DNA methylation dataset using the K-means clustering technique. This function has two arguments <code>File</code> (any matrices) and <code>NCluster</code> ( number of clusters for K-Means clustering).</p>
<h3 id="usage_3">Usage</h3>
<pre><code class="language-r">data &lt;- data.frame(t(Data2[1:100,]))
data &lt;- cbind(rownames(data), data)
colnames(data)[1] &lt;- &quot;ID&quot;
TSNEPlot(File = data, NCluster = 4)
</code></pre>
<p>An R window will appear with a 3D projection of the t-SNE result. The plot object can be saved with the next line of code (uncomment).</p>
<pre><code class="language-r"># rgl.snapshot('tsne3d.png', fmt = 'png')
</code></pre>
<h2 id="input-file-for-similarity-network-fusion-snf">Input file for similarity network fusion (SNF)</h2>
<p>Using <code>ReadSNFData</code> function, one can read files (any matrices with CSV or TSV format) and feed them into the similarity network fusion (SNF) function (from the SNFtools package). Please uncomment the following lines and run the function.</p>
<h3 id="usage_4">Usage</h3>
<pre><code class="language-r"># data(Data2) # Gene expression 
# Data2 &lt;- cbind(rownames(Data2), Data2)
# colnames(Data2)[1] &lt;- &quot;ID&quot;
# write.csv(Data2, &quot;Data2.csv&quot;, row.names = FALSE)
# Data2 &lt;- ReadSNFData(File = &quot;Data2.csv&quot;)
</code></pre>
<h2 id="similarity-network-fusion-snf">Similarity network fusion (SNF)</h2>
<p>The <code>SimilarityNetworkFusion</code> is a function to perform SNF function (from SNFtool package) and output clusters.</p>
<h3 id="usage_5">Usage</h3>
<pre><code class="language-r">data(RLabels) # Real labels
data(Data2) # Methylation
data(Data3) # Gene expression
snf &lt;- SimilarityNetworkFusion(Files = list(Data2, Data3),
                               NNeighbors  = 13,
                               Sigma = 0.75,
                               NClusters = 4,
                               CLabels = c(&quot;Group4&quot;, &quot;SHH&quot;, &quot;WNT&quot;, &quot;Group3&quot;),
                               RLabels = RLabels,
                               Niterations = 60)
</code></pre>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAbAAAAH4CAMAAAAcm0gZAAAAKlBMVEUAAAB9ACWiBwbDIgDhPADtYgDyhAD1oQD3ujz40HT75Jr/9Lf//8j///9YbNX0AAAACXBIWXMAAAsSAAALEgHS3X78AAARu0lEQVR4nO2d4ZriKhJAs7t3Zlq9vv/r7jdalU6VFTqEAEHP+eEMhCCpk6hQxp7uMBRT7wFAHggbDIQNBsIGA2GDgbDBQNhgIGwwEDYYCBsMhA0GwgYDYYOBsMFA2GAgbDAQNhgIGwyEDQbCBgNhg4GwwUDYYCBsMBA2GAgbDIQNBsIGA2GDgbDBQNhgIGwwEDYYCBsMhA0GwgYDYYOBsMFA2GAgbDCewm5P/hVuKa5PtK0UXQ+2o39Drpa4UWVudoRS+i18Pfn15CLYRlorjWSXP4IUpa0+G8J2g7AwHnGwEIawbBAWxiMOFsIQlg3CwnjEwULYYcJkDNaFIhv9Iafkutrkni11rQn7snG2wf+yGrX4zxMbuqsNgJ4ICNtNX2HTNM3/TE8Zjxp5kKrH/yZ9QFg/YdPyn4ePh7L5YVa3aIewvsIe15ZYEUnyoFffLEnrEdZTmF5I38Kmuey3L/5FmC12FDYtNiDsla7C9MVw8ZI4f7ZYvhBO7spD2Hgf6+Vp4nDLxouMzLWVo4lHGNfa/nSS2UWYF2cGGB/4JW4rh+is2jhoLcIQhjCEIQxhCEMYwhBWQ5h0GA8/HoO2lfGGe27D9tAWHYOt1fmSjMxOx/TclI1aa9u62Mku2hZhu+krLCO9Mj8grJ+wnPTKco0RYf2EbU6vzFcZwroK25xemS89hJ1L2Fp6BWF2vLa27YeOjekVeQjTK3badLXEbZPxiIvnFSaZk19ybFafKhIXvy3Snxalkeyp5/zrh/XtIAxhr/GIiwhDWDYIC+MRFxGGsGwQFsYjLiIMYdm8lzB3UNafO+RkPNLBSjaqjHtya0inyk6YIFJ1MmxNaSPpUL4DpyfCHPzc1fq1r2ojrImw7NX6n1Y63EEh7Hhheav1XGHdheWt1iPsfMKSq/UIO8WHjozVeoT1FbaHUJg8mzXkvqTmDtkFIAf7bC11zUfqJoFuUuVM/THoUVi5ulV2sdOxC8J2gzCEIawmCEMYwmqCMISdTtimeZhVEre9xR02xY/wiVWiEr4sVtQfmWI5NdJINuqJgLDd9BW242YIhHUUlpNeMUIR1k3Y9t+akqsQYZ2vsK3pFfWLsJMJW0uvzA8I6yfsuJshENZG2B6MMEX6dRNG2WhLaaxqV7yE50VB2HfseQ2f1TmJxyttVJhslJI7yV1/CEMYwhCGMIQhDGEIQ1gNYUkJOvy4UWhTSQqL2SPsaHT4En0XbtnoZlwWp9G1RdjBtBK2/7emEGZoJCw7vTIvNiLM0k5YXnrley0RYYZ2wjLTK/NqPcIMHYVtTK8gzNBI2HHpFVuM1biD2xGWM5i6hYf6y+ZK3IkrRTdZ/R0iu4i+anevIAxhiWAhDGHZY7BFhCWDhTCEZY/BFhGWDBbCEJY9Bls8vzA7FEXmuTqkLY1crZspu0a2bU9vjjibpZNhqbUl9zdoLharb/5xsJzVev0fwgIaCctarV+uVyHM0U7Y3tV6hBkaXmFbV+un+eUTYa90FLa+Wr94jUSYo5GwrNX6Oy+J67QStodwHmYD68Nu28rGHWG5xR025RqeLe5UktmTO3HFkJt4uftU7C5ai7DdICwDhCEsG4RlgDCEZYOwDBC2X1iXYMVcLXLE15CLxdUm9/nPk/8JUozPnU1nVHzi3kIQNqiwvJsh7ouVjiMjXsgHCctJr8wbENZV2N7fmjoy4oV8lrDt6RWusHMKW0uvIOwEwgpuhjgy4oV8krA9nE6YC7dMOVcUCbZR3CYW9l9BImm9HTCXjztCGML6gTCEIawmCEMYwmqCsB3C5NjKx7sDDawcqsT3lkSEuVrbkRMm8y9tK96k9moD4KIeDnBlhLYHBWGDCtv/l9LjoDUR5PkgYTnplbt+sx5hXYVl3r3CFdZb2Ob0ytwcYecStpZeQdgJhOWnV9aE9eRqERcali+D3k9iG32FxYtFQufSK0LtY7wXgDCElYEwhCGsJghDGMJqgjCEIawmv0NUzY5a18bNd0NqH+Mc/PybIc73RdIPEpazWj/XIayrsMybIc74zd/PErZ5tX6+9BB2LmFrq/UIO4Gw97gZ4pOE7eHswly4fxm0VqZasWsryv1U7zWk9jEiDGH9QBjCEFYThCEMYTVB2GDC4rlVjIqzwrRWOrKi9AyQXVSj9Vf7GBE2qLCM9MqUvBmiJx8kLCe9MutEWFdh29Mrcz3Cul5hW9Mry2V7hJ1I2Fp6BWEnEJaXXkFYf2F7OJ0w50Ti66fKZuNKPkxcxCeAmzgjbDcIQxjCaoIwhCGsJghDGMJq8sfyO4VqlHBL7c2acqqlNp6HCbWPEWGDCsv+rakzr9Z/gLCc9Mq8JoywrsI2p1f0KkRY5ytsa3pFdSHsZMLW0ivzJoR1/9DxJnev/PMkNhVbleAnHSuiRoWZb7kxD8sBYQhDWE0QhjCE1QRhCENYTRC2Q5gcRe1xh6TjLPNdCbcKs6Z0+mt3UeJa2296hOXRmYOfuVo/rS1NISzFYcKyV+u1AmHdhOWt1q9+8xdhKQ69wjJX63kPO5uw9Go9wroK27Faj7CuwvZwunlYPMVyRUFnUXYXd/dKnA+zqvnW1H4QhjCE1QRhCENYTRCGMITVJBamSPTtxEtnXvEudk8VJxMvJ2zU9EpPPknYB9wM8VbCctIrRijCBvitqUFvhng3YZvTK3e99BB2LmFr6RWEnUDYe9wM8UnC9nA6YS7OEthf1pSbDNtUF8KagjCEIawmCEMYwmqCMIQhrCYuzjK3uoWmViZrVlHcoRjSfge/e6UnnyQsJ73y3HjCH1b5IGFZ6ZVlBcK6CctLr3yLrD3CDD5L2Ob0yjT/OhjCTiVsPb2yqERY1w8d29IrowiTgKqaLTMupyaWKx2Jt4st1j7GewEIQ1gZCEMYwmqCMIQhrCYIQxjCauIyX0kXToL8npirjZEp8hkmzttX60/5vcQPEvYeq/WfJSxztZ4rrLewzNV6rrDTCUuv1iPsDB86clbreUnsKmwPpxOmIYyzWUlhbqMtxsIUvua2G4QhDGE1QRjCEFYThCHsDYXJePVQJdkgpZvFbvwjx+hCaNMVcb9ftpF0XztYOdghuTmbq0324NogrBK1heX91tR9cTMEwkIqC8tJr8zLjAhLUF/Y3pshEBbS4ArbnF5ZViJshQ7C1tIrCNtCZWEFf8oDYSG1he3BCHPTyFuIhFva6F84uVikI1EST09VmPQrG+sq2IZTE8fBHtPFHoViz3mNA8IOBmEIQ1hNEIYwhNUEYQirI8xJcOG2cXYTr2tIUpge+Zeht6y/xIaU5OnssOckwirRSlhueuVl8RdhTxoJ25FeQVhIO2F7715BmKGdsLz0ykvGGWFPOgpLp1d4SYxpJCw/veKF6XfA7LNdbJwFZ0pK7stkrt94+LaH3rKWyEBt6RbXhnuuTC7vBSAsAcIQ9gBhlUAYwh4grBIIQ9gDhFXivMLs3FUnuheL1MZtXa3s4mrjE8G2rewgCxtud4Kl/YVov3Pwc1fr57VEhIVUFpa9Wj+vfyAspr6w3JshEJakwRWWtVr/feUhLKSDsPRqvTZAWExlYdmr9d/5MISF1Ba2h6cwGYObPrmR2dqceZj2IE5kowvAW87DbAh1K8IqgTCEPUBYJRCGsAcIqwTCEPagXJibNtliHNivkIvFqr6E3btd26jYhhyxLV2thDTxLgirRG1hGemVefUDYQkqC8tJr9xtegVhIfWFbU+v6P8QlqDBFbY1vWJeMhG2Qgdha+kVhG2hsrC89Iq+0SFsndrC9mCE6WTYxlmfzhZ/2+mvmzHbHtw8Op44i7euhhxWmDsLHXbPdBuEVQJhCHuAsEogDGEPEFYJhCHsQbkwGVkszCEbXT7MirqFhvTuFSfMni0NffyIjbMLgG3TJ4GJMEttYTvSKxPCElQWlp1e+f6qNsJC6gvbm15BWEiDKywrvTLLQ1hMB2HJ9ArCfqCysPL0ihMWp1ecXDuk3xbpQdva4vmFycikdLNs6sFGR7kXgLAECEMYwmqCMIQhrCYIQxjCanJeYU6G7V5rbbjdNDK2ukm5nUfXC/923EAvllsKt4u9ArTDOfhZvzWl02uEvdBIWM5q/dwWYQHthG1erddVLIRFtBO2ebV+EokTwgI6CltbrZ+LCAtoJKzgL0MgzNBK2B6ewtx8yYbbjddJtSNzEpLzMMX20FvWEjlEW7rZ2pweFIRVAmEIe4CwSiAMYQ8QVgmEIexBuTDrYuVPH0qtlNxsyrUNN7pp2Zf1J6VDI76TWxIbnZXz2LZ18UXYwbQSlp1e8d9LRNiTRsKy0yvzghXCLO2EZaZXEBbT8ArLSq98X3kIM3QUlk6vaAOEWRoJK/hTHggztBK2h6cwidnFOrEBXbkJyUq4hCP0A7bYQ+wt6y83GwB3UEmb7mQUXCOEHQzCEIawmiAMYQirCcIQVleYPKsVdYu92eDHtTe3j21zPmHuaKy+r/AQ9SicsNgxwg6mlbDc9Mr3SgfCDI2E5aRX5sYIC2gnLPcvpU+8JEY0vMK2pldMJcIcHYWtpVcQlqKRsPy7V2ZhbsZlZw+u1g3JHo1V4v8YiK11Pciz9Jb1Fw2sDNvVxhJcI9uDOzXvBSAsAGEIQ1hNEIYwhNUEYQhDWE3OL+waduzCbcfrBxHi/Nld9BfErMaCAO/YU4/C1l5iJ3bc8bHFUXH9zcHPWK3XKTTC+gnLWq3XlSmEdRWWu1qPsN5X2NbV+mnLz+8hrL2w9dV6s/iLsB7Cslbrny+g6V9zQ1hlYXsw8zDtUJ5VZkbxyHR25ooW6VfV2GLc4R5htbAngpOwowcFYZVAGMIeIKwSCEPYA4RVAmEIe1AuTPr9sjMvKTkJEmbX1gb/Gjpe+bUxy6ER38nNxkNKOsLbBlwc3FaEHUwrYfvTKwgzNBJWkF5BmKGdsLz0CsJWaHiFZaVXELZCR2HJ9ArCVmgkLDu9woeOFVoJ28NUsjO0B2GDgbDBQNhgIGwwEDYYup6l35+S0tLk5KTarS//n2x5vV+/q32a1z3tGKbUAH3N+gD9ZtOvbftybNufJvWUmXzPpL+jt1h8fBZ9kBc10+SO0bo2Rd9vsGuyIzfqaX3j+tP4fv0YXvpNHFtmHI5hRdjdn2j2oFbPnmlKHKPf01+M90RgX8dgTwQ33mWAnE17lfvL2vTrX2kSwl46Sg2/hPgl8X5fHYrfmn4bnPyLSuIlccMwv8sv+szGVE/JyKW37n6a414R+dAxGtPLf15Kyffe9Ltr8oPED0V/Lq0PKTnAHz7MvLwur48hKw7JMZQg72H2jTnrg0Tq3fXnDxKr/boxpIa0od9DxpAbh9WwlKAduRPLHGP6g0Tq3dUWc4KVvljtkH7+NJf6lLh5DMnPIDljKGLDS+LLLm7Olnh3fSke9JKYHNFPQ0p+6EiPIUHOGEo4sCtowaSP299AaVvYtggVlvkGStuCtkWIMPsGmp7h07awbREHXajQCoQNBsIGY34PS6x0pIq0zW9bxGT+uUeldJG2+W1L4CVxMBA2GJM+nmw14J3bFqHCzCTwBKsB79y2CIQNKSyZMvkhc0Db7LYlHNQNtGK+wswVbC/gZJG2+W1LmPRx+aVIU0oXaZvftohZ2OJzpy2li7TNb1vEtOjR9n/fVqRtftsSjukFmqFX2GTOlsmeHqkibfPbljDp48nemN+5bRGzsJO9Mb9z2yKmRY+2//u2Im3z25ZwTC/QDHlJPF0a/Z3bFjGZf+5RKV2kbX7bErSn8y1pv3PbArjCOrQtgfew9m2LOE49NAFhgzHpo52Xm3fTVJG2+W2LUGHT4mI7wxdV3rltEQgbUphb/k/nBmhb2raEY3qBZiBsMFTYy0LKfWuRtvltC5jk8WxvzO/ctojp+3EyddN9U5G2+W2LQNiQwk6YgXjntiUc1A20AmGDgbDBQNhgIGwwEDYYCBsMhA0GwgYDYYOBsMFA2GAgbDAQNhgIGwyEDQbCBgNhg4GwwUDYYCBsMBA2GAgbDIQNBsIGA2GDgbDBQNhgIGwwEDYYCBsMhA0GwgYDYYOBsMFA2GAgbDD+D8iXuhELnRlRAAAAAElFTkSuQmCC" alt="plot of chunk unnamed-chunk-7" /></p>
<pre><code class="language-r">snf
#&gt;  [1] SHH    Group3 Group4 Group4 Group4 SHH    SHH    Group3 Group4 SHH   
#&gt; [11] WNT    SHH    SHH    WNT    SHH    WNT    Group3 Group3 Group3 Group4
#&gt; [21] Group4 Group3 Group3 Group3 Group4 Group4 Group4 Group3 Group3 SHH   
#&gt; [31] SHH    SHH    SHH    SHH    Group4 Group3 SHH    Group4 Group4 Group3
#&gt; [41] Group4 Group4 WNT    Group3 Group4 Group4 Group4 Group4 SHH    Group4
#&gt; Levels: Group4 SHH WNT Group3
</code></pre>
<p>This function has several arguments as follow:</p>
<ul>
<li><code>Files</code>   A list of data frames created using the ReadSNFData function.</li>
<li><code>NNeighbors</code>    The number of nearest neighbors.</li>
<li><code>Sigma</code>     The variance for local model.</li>
<li><code>NClusters</code>   The number of clusters.</li>
<li><code>CLabels</code>   A string vector to name the clusters. Optional.</li>
<li><code>RLabels</code>   The actual label of samples to calculate the Normalized Mutual Information (NMI) score. Optional.</li>
<li><code>Niterations</code>   The number of iterations for the diffusion process.</li>
</ul>
<h2 id="support-vector-machine-model">Support vector machine model</h2>
<p>The <code>SupportVectorMachineModel</code> is a function to train a support vector machine model to classify medulloblastoma subgroups using DNA methylation beta values (Illumina Infinium HumanMethylation450). Prediction is followed by training if new data is provided.</p>
<p>Model metrics, including accuracy, precision, sensitivity F1-Score, specificity, and AUC_average can be calculated for the test dataset using the <code>ModelMetrics</code> function, which calculates the average of the above parameters from the result of the <code>ConfusionMatrix</code> function.</p>
<p>The prediction result on new data can be accessed through the <code>NewDataPredictionResult</code> function, which calculates every predictionâ€™s mode across the number of cross-validation folds.</p>
<h3 id="usage_6">Usage</h3>
<pre><code class="language-r">set.seed(1234)
fac &lt;- ncol(Data1)
NewData &lt;- sample(data.frame(t(Data1[,-fac])),10)
NewData &lt;- cbind(rownames(NewData), NewData)
colnames(NewData)[1] &lt;- &quot;ID&quot;

svm &lt;- SupportVectorMachineModel(SplitRatio = 0.8, 
                                 CV = 10, 
                                 NCores = 1, 
                                 NewData = NewData)
ModelMetrics(Model = svm)
#&gt; $ConfusionMatrix
#&gt;         y_pred
#&gt;          Group3 SHH WNT Group4
#&gt;   Group3     26   0   0      0
#&gt;   SHH         0  41   0      0
#&gt;   WNT         1   0  37      0
#&gt;   Group4      1   0   0     59
#&gt; 
#&gt; $ModelPerformance
#&gt;        Accuracy Precision Sensitivity F1_Score Specificity AUC_average
#&gt; Group3    0.988     0.932       0.996    0.963       0.986       0.985
#&gt; SHH       1.000     1.000       1.000    1.000       1.000       0.985
#&gt; WNT       0.995     1.000       0.976    0.988       1.000       0.985
#&gt; Group4    0.993     0.998       0.983    0.990       0.999       0.985
NewDataPredictionResult(Model = svm)
#&gt;            Subgroup
#&gt; GSM2261711   Group3
#&gt; X78             WNT
#&gt; GSM2261640   Group4
#&gt; GSM2261575   Group4
#&gt; X135            WNT
#&gt; GSM2262184   Group3
#&gt; GSM2261613   Group3
#&gt; X130            WNT
#&gt; GSM2261922   Group4
#&gt; GSM2261980   Group3
</code></pre>
<p>This function has the following arguments:</p>
<ul>
<li><code>SplitRatio</code>    Train and test split ratio. A value greater or equal to zero and less than one.</li>
<li><code>CV</code>    The number of folds for cross-validation. It should be greater than one.</li>
<li><code>NCores</code>    The number of cores for parallel computing.</li>
<li><code>NewData</code>    A methylation beta values input from the ReadMethylFile function.</li>
</ul>
<h2 id="k-nearest-neighbor-model">K nearest neighbor model</h2>
<p>The <code>KNearestNeighborModel</code> is a function to train a K nearest neighbor model to classify medulloblastoma subgroups using DNA methylation beta values.</p>
<h3 id="usage_7">Usage</h3>
<pre><code class="language-r">set.seed(1234)
fac &lt;- ncol(Data1)
NewData &lt;- sample(data.frame(t(Data1[,-fac])),10)
NewData &lt;- cbind(rownames(NewData), NewData)
colnames(NewData)[1] &lt;- &quot;ID&quot;

knn &lt;- KNearestNeighborModel(SplitRatio = 0.8, 
                             CV = 10, 
                             K = 3, 
                             NCores = 1, 
                             NewData = NewData)
ModelMetrics(Model = knn)
#&gt; $ConfusionMatrix
#&gt;         knnclass_pred
#&gt;          Group3 SHH WNT Group4
#&gt;   Group3     25   0   0      1
#&gt;   SHH         0  41   0      0
#&gt;   WNT         0   0  38      0
#&gt;   Group4      1   0   0     59
#&gt; 
#&gt; $ModelPerformance
#&gt;        Accuracy Precision Sensitivity F1_Score Specificity AUC_average
#&gt; Group3    0.993     0.981       0.973    0.977       0.996       0.985
#&gt; SHH       0.999     1.000       0.997    0.999       1.000       0.985
#&gt; WNT       0.999     0.997       1.000    0.999       0.999       0.985
#&gt; Group4    0.993     0.988       0.991    0.990       0.993       0.985
NewDataPredictionResult(Model = knn)
#&gt;            Subgroup
#&gt; GSM2261711   Group3
#&gt; X78             WNT
#&gt; GSM2261640   Group4
#&gt; GSM2261575   Group4
#&gt; X135            WNT
#&gt; GSM2262184   Group3
#&gt; GSM2261613   Group3
#&gt; X130            WNT
#&gt; GSM2261922   Group4
#&gt; GSM2261980   Group3
</code></pre>
<p>This function has the following arguments:</p>
<ul>
<li><code>SplitRatio</code>    Train and test split ratio. A value greater or equal to zero and less than one.</li>
<li><code>CV</code>    The number of folds for cross-validation. It should be greater than one.</li>
<li><code>K</code>   The number of nearest neighbors.</li>
<li><code>NCores</code>    The number of cores for parallel computing.</li>
<li><code>NewData</code>    A methylation beta values input from the ReadMethylFile function.</li>
</ul>
<h2 id="random-forest-model">Random forest model</h2>
<p>The <code>RandomForestModel</code> is a function to train a random forest model to classify medulloblastoma subgroups using DNA methylation beta values.</p>
<h3 id="usage_8">Usage</h3>
<pre><code class="language-r">set.seed(1234)
fac &lt;- ncol(Data1)
NewData &lt;- sample(data.frame(t(Data1[,-fac])),10)
NewData &lt;- cbind(rownames(NewData), NewData)
colnames(NewData)[1] &lt;- &quot;ID&quot;

rf &lt;- RandomForestModel(SplitRatio = 0.8, 
                        CV = 10, 
                        NTree = 100, 
                        NCores = 1, 
                        NewData = NewData)
ModelMetrics(Model = rf)
#&gt; $ConfusionMatrix
#&gt;         y_pred
#&gt;          Group3 SHH WNT Group4
#&gt;   Group3     26   0   0      0
#&gt;   SHH         0  41   0      0
#&gt;   WNT         0   0  38      0
#&gt;   Group4      0   0   0     60
#&gt; 
#&gt; $ModelPerformance
#&gt;        Accuracy Precision Sensitivity F1_Score Specificity AUC_average
#&gt; Group3    0.999     1.000       0.996    0.998       1.000       0.998
#&gt; SHH       1.000     1.000       1.000    1.000       1.000       0.998
#&gt; WNT       1.000     1.000       1.000    1.000       1.000       0.998
#&gt; Group4    0.999     0.998       1.000    0.999       0.999       0.998
NewDataPredictionResult(Model = rf)
#&gt;            Subgroup
#&gt; GSM2261711   Group3
#&gt; X78             WNT
#&gt; GSM2261640   Group4
#&gt; GSM2261575   Group4
#&gt; X135            WNT
#&gt; GSM2262184   Group3
#&gt; GSM2261613   Group3
#&gt; X130            WNT
#&gt; GSM2261922   Group4
#&gt; GSM2261980   Group3
</code></pre>
<p>This function has the following arguments:</p>
<ul>
<li><code>SplitRatio</code>    Train and test split ratio. A value greater or equal to zero and less than one.</li>
<li><code>CV</code>    The number of folds for cross-validation. It should be greater than one.</li>
<li><code>NTree</code>   The number of trees to be grown.</li>
<li><code>NCores</code>    The number of cores for parallel computing.</li>
<li><code>NewData</code>  A methylation beta values input from the ReadMethylFile function.</li>
</ul>
<h2 id="xgboost-model">XGBoost model</h2>
<p>The <code>XGBoostModel</code> is a A function to train an XGBoost model to classify medulloblastoma subgroups using DNA methylation beta values.</p>
<h3 id="usage_9">Usage</h3>
<pre><code class="language-r">set.seed(1234)
fac &lt;- ncol(Data1)
NewData &lt;- sample(data.frame(t(Data1[,-fac])),10)
NewData &lt;- cbind(rownames(NewData), NewData)
colnames(NewData)[1] &lt;- &quot;ID&quot;

xgboost &lt;- XGBoostModel(SplitRatio = 0.8, 
                        CV = 10, 
                        NCores = 1, 
                        NewData = NewData)
#&gt; [1]	train-mlogloss:0.390594 
#&gt; Will train until train_mlogloss hasn't improved in 10 rounds.
#&gt; 
#&gt; [2]	train-mlogloss:0.177861 
#&gt; [3]	train-mlogloss:0.087035 
#&gt; [4]	train-mlogloss:0.043112 
#&gt; [5]	train-mlogloss:0.022536 
#&gt; [6]	train-mlogloss:0.012486 
#&gt; [7]	train-mlogloss:0.007278 
#&gt; [8]	train-mlogloss:0.004395 
#&gt; [9]	train-mlogloss:0.002879 
#&gt; [10]	train-mlogloss:0.002457 
#&gt; [1]	train-mlogloss:0.388419 
#&gt; Will train until train_mlogloss hasn't improved in 10 rounds.
#&gt; 
#&gt; [2]	train-mlogloss:0.177664 
#&gt; [3]	train-mlogloss:0.085746 
#&gt; [4]	train-mlogloss:0.043333 
#&gt; [5]	train-mlogloss:0.022637 
#&gt; [6]	train-mlogloss:0.012444 
#&gt; [7]	train-mlogloss:0.007140 
#&gt; [8]	train-mlogloss:0.004413 
#&gt; [9]	train-mlogloss:0.002823 
#&gt; [10]	train-mlogloss:0.002431 
#&gt; [1]	train-mlogloss:0.388072 
#&gt; Will train until train_mlogloss hasn't improved in 10 rounds.
#&gt; 
#&gt; [2]	train-mlogloss:0.176992 
#&gt; [3]	train-mlogloss:0.085394 
#&gt; [4]	train-mlogloss:0.043119 
#&gt; [5]	train-mlogloss:0.022323 
#&gt; [6]	train-mlogloss:0.012245 
#&gt; [7]	train-mlogloss:0.006953 
#&gt; [8]	train-mlogloss:0.004304 
#&gt; [9]	train-mlogloss:0.002808 
#&gt; [10]	train-mlogloss:0.002544 
#&gt; [1]	train-mlogloss:0.386945 
#&gt; Will train until train_mlogloss hasn't improved in 10 rounds.
#&gt; 
#&gt; [2]	train-mlogloss:0.175823 
#&gt; [3]	train-mlogloss:0.085418 
#&gt; [4]	train-mlogloss:0.042969 
#&gt; [5]	train-mlogloss:0.022146 
#&gt; [6]	train-mlogloss:0.012049 
#&gt; [7]	train-mlogloss:0.006975 
#&gt; [8]	train-mlogloss:0.004246 
#&gt; [9]	train-mlogloss:0.002766 
#&gt; [10]	train-mlogloss:0.002319 
#&gt; [1]	train-mlogloss:0.387957 
#&gt; Will train until train_mlogloss hasn't improved in 10 rounds.
#&gt; 
#&gt; [2]	train-mlogloss:0.177210 
#&gt; [3]	train-mlogloss:0.085601 
#&gt; [4]	train-mlogloss:0.043317 
#&gt; [5]	train-mlogloss:0.022903 
#&gt; [6]	train-mlogloss:0.012530 
#&gt; [7]	train-mlogloss:0.007282 
#&gt; [8]	train-mlogloss:0.004478 
#&gt; [9]	train-mlogloss:0.002934 
#&gt; [10]	train-mlogloss:0.002514 
#&gt; [1]	train-mlogloss:0.390082 
#&gt; Will train until train_mlogloss hasn't improved in 10 rounds.
#&gt; 
#&gt; [2]	train-mlogloss:0.177320 
#&gt; [3]	train-mlogloss:0.085780 
#&gt; [4]	train-mlogloss:0.043300 
#&gt; [5]	train-mlogloss:0.022592 
#&gt; [6]	train-mlogloss:0.012513 
#&gt; [7]	train-mlogloss:0.007264 
#&gt; [8]	train-mlogloss:0.004434 
#&gt; [9]	train-mlogloss:0.002923 
#&gt; [10]	train-mlogloss:0.002552 
#&gt; [1]	train-mlogloss:0.391327 
#&gt; Will train until train_mlogloss hasn't improved in 10 rounds.
#&gt; 
#&gt; [2]	train-mlogloss:0.178573 
#&gt; [3]	train-mlogloss:0.086585 
#&gt; [4]	train-mlogloss:0.043456 
#&gt; [5]	train-mlogloss:0.022623 
#&gt; [6]	train-mlogloss:0.012235 
#&gt; [7]	train-mlogloss:0.007101 
#&gt; [8]	train-mlogloss:0.004310 
#&gt; [9]	train-mlogloss:0.002876 
#&gt; [10]	train-mlogloss:0.002484 
#&gt; [1]	train-mlogloss:0.387270 
#&gt; Will train until train_mlogloss hasn't improved in 10 rounds.
#&gt; 
#&gt; [2]	train-mlogloss:0.176343 
#&gt; [3]	train-mlogloss:0.085810 
#&gt; [4]	train-mlogloss:0.042979 
#&gt; [5]	train-mlogloss:0.022061 
#&gt; [6]	train-mlogloss:0.011726 
#&gt; [7]	train-mlogloss:0.006691 
#&gt; [8]	train-mlogloss:0.004024 
#&gt; [9]	train-mlogloss:0.002633 
#&gt; [10]	train-mlogloss:0.002421 
#&gt; [1]	train-mlogloss:0.385785 
#&gt; Will train until train_mlogloss hasn't improved in 10 rounds.
#&gt; 
#&gt; [2]	train-mlogloss:0.175053 
#&gt; [3]	train-mlogloss:0.084670 
#&gt; [4]	train-mlogloss:0.042764 
#&gt; [5]	train-mlogloss:0.022229 
#&gt; [6]	train-mlogloss:0.011936 
#&gt; [7]	train-mlogloss:0.006921 
#&gt; [8]	train-mlogloss:0.004281 
#&gt; [9]	train-mlogloss:0.002846 
#&gt; [10]	train-mlogloss:0.002425 
#&gt; [1]	train-mlogloss:0.388743 
#&gt; Will train until train_mlogloss hasn't improved in 10 rounds.
#&gt; 
#&gt; [2]	train-mlogloss:0.176686 
#&gt; [3]	train-mlogloss:0.086097 
#&gt; [4]	train-mlogloss:0.043017 
#&gt; [5]	train-mlogloss:0.022624 
#&gt; [6]	train-mlogloss:0.012143 
#&gt; [7]	train-mlogloss:0.007023 
#&gt; [8]	train-mlogloss:0.004200 
#&gt; [9]	train-mlogloss:0.002729 
#&gt; [10]	train-mlogloss:0.002506
ModelMetrics(Model = xgboost)
#&gt; $ConfusionMatrix
#&gt;         y_pred
#&gt; y_truth  Group3 Group4 SHH WNT
#&gt;   Group3     24      1   0   1
#&gt;   Group4      2     58   0   0
#&gt;   SHH         0      0  41   0
#&gt;   WNT         3      0   0  35
#&gt; 
#&gt; $ModelPerformance
#&gt;        Accuracy Precision Sensitivity F1_Score Specificity AUC_average
#&gt; Group3    0.979     0.896       0.981    0.936       0.978       0.968
#&gt; Group4    0.993     0.990       0.991    0.991       0.994       0.968
#&gt; SHH       0.998     1.000       0.990    0.995       1.000       0.968
#&gt; WNT       0.983     0.992       0.934    0.962       0.998       0.968
NewDataPredictionResult(Model = xgboost)
#&gt;            Subgroup
#&gt; GSM2261711   Group3
#&gt; X78             WNT
#&gt; GSM2261640   Group4
#&gt; GSM2261575   Group4
#&gt; X135            WNT
#&gt; GSM2262184   Group3
#&gt; GSM2261613   Group3
#&gt; X130            WNT
#&gt; GSM2261922   Group4
#&gt; GSM2261980   Group3
</code></pre>
<p>This function has the following arguments:</p>
<ul>
<li><code>SplitRatio</code>    Train and test split ratio. A value greater or equal to zero and less than one.</li>
<li><code>CV</code>    The number of folds for cross-validation. It should be greater than one.</li>
<li><code>NCores</code>    The number of cores for parallel computing.</li>
<li><code>NewData</code>  A methylation beta values input from the ReadMethylFile function.</li>
</ul>
<h2 id="linear-discriminant-analysis-model">Linear discriminant analysis model</h2>
<p>The <code>LinearDiscriminantAnalysisModel</code> is a function to train a linear discriminant analysis model to classify medulloblastoma subgroups using DNA methylation beta values.</p>
<h3 id="usage_10">Usage</h3>
<pre><code class="language-r">set.seed(1234)
fac &lt;- ncol(Data1)
NewData &lt;- sample(data.frame(t(Data1[,-fac])),10)
NewData &lt;- cbind(rownames(NewData), NewData)
colnames(NewData)[1] &lt;- &quot;ID&quot;

lda &lt;- LinearDiscriminantAnalysisModel(SplitRatio = 0.8, 
                                       CV = 10, 
                                       NCores = 1, 
                                       NewData = NewData)
ModelMetrics(Model = lda)
#&gt; $ConfusionMatrix
#&gt;         y_pred
#&gt;          Group3 SHH WNT Group4
#&gt;   Group3     22   1   0      3
#&gt;   SHH         0  41   0      0
#&gt;   WNT         0   0  38      0
#&gt;   Group4      4   1   0     55
#&gt; 
#&gt; $ModelPerformance
#&gt;        Accuracy Precision Sensitivity F1_Score Specificity AUC_average
#&gt; Group3    0.941     0.778       0.889    0.828       0.951        0.91
#&gt; SHH       0.994     0.991       0.985    0.988       0.997        0.91
#&gt; WNT       0.993     0.986       0.981    0.984       0.996        0.91
#&gt; Group4    0.945     0.949       0.893    0.920       0.973        0.91
NewDataPredictionResult(Model = lda)
#&gt;            Subgroup
#&gt; GSM2261711   Group3
#&gt; X78             WNT
#&gt; GSM2261640   Group4
#&gt; GSM2261575   Group4
#&gt; X135            WNT
#&gt; GSM2262184   Group3
#&gt; GSM2261613   Group3
#&gt; X130            WNT
#&gt; GSM2261922   Group4
#&gt; GSM2261980   Group3
</code></pre>
<p>This function has the following arguments:</p>
<ul>
<li><code>SplitRatio</code>    Train and test split ratio. A value greater or equal to zero and less than one.</li>
<li><code>CV</code>    The number of folds for cross-validation. It should be greater than one.</li>
<li><code>NCores</code>    The number of cores for parallel computing.</li>
<li><code>NewData</code>  A methylation beta values input from the ReadMethylFile function.</li>
</ul>
<h2 id="naive-bayes-model">Naive bayes model</h2>
<p>The <code>NaiveBayesModel</code> is a function to train a Naive Bayes model to classify medulloblastoma subgroups using DNA methylation beta values.</p>
<h3 id="usage_11">Usage</h3>
<pre><code class="language-r">set.seed(1234)
fac &lt;- ncol(Data1)
NewData &lt;- sample(data.frame(t(Data1[,-fac])),10)
NewData &lt;- cbind(rownames(NewData), NewData)
colnames(NewData)[1] &lt;- &quot;ID&quot;

nb &lt;- NaiveBayesModel(SplitRatio = 0.8, 
                      CV = 10, 
                      Threshold = 0.8, 
                      NCores = 1, 
                      NewData = NewData)
ModelMetrics(Model = nb)
#&gt; $ConfusionMatrix
#&gt;         y_pred
#&gt;          Group3 SHH WNT Group4
#&gt;   Group3     26   0   0      0
#&gt;   SHH         0  41   0      0
#&gt;   WNT         3   0  35      0
#&gt;   Group4      1   0   0     59
#&gt; 
#&gt; $ModelPerformance
#&gt;        Accuracy Precision Sensitivity F1_Score Specificity AUC_average
#&gt; Group3    0.974     0.859       1.000    0.924       0.969       0.971
#&gt; SHH       1.000     1.000       1.000    1.000       1.000       0.971
#&gt; WNT       0.984     1.000       0.928    0.963       1.000       0.971
#&gt; Group4    0.990     1.000       0.972    0.986       1.000       0.971
NewDataPredictionResult(Model = nb)
#&gt;            Subgroup
#&gt; GSM2261711   Group3
#&gt; X78             WNT
#&gt; GSM2261640   Group4
#&gt; GSM2261575   Group4
#&gt; X135            WNT
#&gt; GSM2262184   Group3
#&gt; GSM2261613   Group3
#&gt; X130            WNT
#&gt; GSM2261922   Group4
#&gt; GSM2261980   Group3
</code></pre>
<p>This function has the following arguments:</p>
<ul>
<li><code>SplitRatio</code>    Train and test split ratio. A value greater or equal to zero and less than one.</li>
<li><code>CV</code>    The number of folds for cross-validation. It should be greater than one.</li>
<li><code>Threshold</code>   The threshold for deciding class probability. A value greater or equal to zero and less than one.</li>
<li><code>NCores</code>    The number of cores for parallel computing.</li>
<li><code>NewData</code>  A methylation beta values input from the ReadMethylFile function.</li>
</ul>
<h2 id="artificial-neural-network-model">Artificial neural network model</h2>
<p>The <code>NeuralNetworkModel</code> is a function to train an artificial neural network model to classify medulloblastoma subgroups using DNA methylation beta values. Please uncomment the following lines and run the function. If it is the first time you run this function, set the InstallTensorFlow parameter to TRUE. It will automatically install the Python and TensorFlow library (version 2.10-cpu) in a virtual environment then set the parameter to FALSE.</p>
<h3 id="usage_12">Usage</h3>
<pre><code class="language-r"># set.seed(1234)
# fac &lt;- ncol(Data1)
# NewData &lt;- sample(data.frame(t(Data1[,-fac])),10)
# NewData &lt;- cbind(rownames(NewData), NewData)
# colnames(NewData)[1] &lt;- &quot;ID&quot;
# ann &lt;- NeuralNetworkModel(Epochs = 100, 
#                           NewData = NewData,
#                           InstallTensorFlow = TRUE)
# ModelMetrics(Model = ann)
# NewDataPredictionResult(Model = ann)
</code></pre>
<p>This function has the following arguments:</p>
<ul>
<li><code>Epochs</code>    The number of epochs.</li>
<li><code>NewData</code>   A methylation beta values input from the ReadMethylFile function.</li>
<li><code>InstallTensorFlow</code>   Logical. Running this function for the first time, you need to install TensorFlow library (V 2.10-cpu). Default is TRUE.</li>
</ul>
</div>
<div class="include-after">
</div>
<script src="https://cdn.jsdelivr.net/npm/prismjs@1.29.0/components/prism-core.min.js" defer></script>
<script src="https://cdn.jsdelivr.net/npm/prismjs@1.29.0/plugins/autoloader/prism-autoloader.min.js" defer></script>
</body>
</html>
